examples/environment-guide/README.md

```diff
@@ -0,0 +1,88 @@
+services on top of both. Accessing the frontend pod will return
+environment information about itself, and a backend pod that it has
+accessed through the service. The goal is to illuminate the
+environment metadata available to running containers inside the
+Kubernetes cluster. The documentation for the kubernetes environment
+is [here](/docs/container-environment.md).
+
+![Diagram](diagram.png)
+
+Prerequisites
+-------------
+This example assumes that you have a Kubernetes cluster installed and
+running, and that you have installed the `kubectl` command line tool
+somewhere in your path.  Please see the [getting
+started](/docs/getting-started-guides) for installation instructions
+for your platform.
+
+Optional: Build your own containers
+-----------------------------------
+The code for the containers is under
+[containers/](containers)
+
+Get everything running
+----------------------
+
+    kubectl create -f ./backend-rc.yaml
+    kubectl create -f ./backend-service.yaml
+    kubectl create -f ./show-rc.yaml
+    kubectl create -f ./show-service.yaml
+
+Query the service
+-----------------
+Use `kubectl describe service show-srv` to determine the public IP of
+your service.
+
+> Note: If your platform does not support external load balancers,
+  you'll need to open the proper port and direct traffic to the
+  internal IP shown for the frontend service with the above command
+
+Run `curl <public ip>:80` to query the service. You should get
+something like this back:
+
+```
+Hostname: show-rc-x09sf
+USER_VAR: important information
+
+Kubernetes environment variables
+BACKEND_SRV_SERVICE_HOST = 10.83.250.240
+BACKEND_SRV_SERVICE_PORT = 5000
+KUBERNETES_RO_SERVICE_HOST = 10.83.240.1
+KUBERNETES_RO_SERVICE_PORT = 80
+KUBERNETES_SERVICE_HOST = 10.83.240.2
+KUBERNETES_SERVICE_PORT = 443
+KUBE_DNS_SERVICE_HOST = 10.83.240.10
+KUBE_DNS_SERVICE_PORT = 53
+SHOW_SRV_SERVICE_HOST = 10.83.242.48
+SHOW_SRV_SERVICE_PORT = 80
+
+Found backend ip: 10.83.250.240 port: 5000
+Response from backend
+Backend Container
+Backend Hostname: backend-rc-fs6ap
+```
+
+First the frontend pods information is printed. The hostname is the
+name of the pod. Next, `USER_VAR` is the name of an environment
+variable set in the [pod definition](show-rc.yaml). Then, the dynamic
+kubernetes environment variables are scanned and printed. These are
+used to find the backend service, named `backend-srv`. Finally, the
+frontend pod queries the backend service and prints the information
+returned. Again the backend pod returns its own hostname aka pod name.
+
+Try running the `curl` command a few times, and notice what
+changes. Firstly, the frontend service is directing your request to
+different frontend pods each time. The frontend pods are always
+contacting the backend through the backend service. This results in a
+different backend pod servicing each request as well.
+
+Cleanup
+-------
+    kubectl delete service show-srv

```

Comment:
If the service were labeled, you could do `kubectl delete rc,service -l type=backend-type`

Alternatively, you could do `kubectl delete rc/show-src service/show-rc`.